feat: unstaged file indicators in pickers

gusjengis opened the pull request:
# Git Status Indicators
All this does is put marks in the picker on any projects that have uncommitted changes. I found myself checking manually and thought this would be a useful feature.

### Telescope
![pr-telescope](https://github.com/user-attachments/assets/b928689f-5e6a-4ea7-8d01-630b5289e821)

### fzf-lua
![pr-fzf-lua](https://github.com/user-attachments/assets/5a52958e-67c7-4ca3-a4ea-afb5ba22463d)

### built-in picker
![pr-builtin](https://github.com/user-attachments/assets/def10c82-1f5b-4bba-8b43-37526387467e)

Open to ideas about how this might be better displayed.

There is currently a slight performance issue with this, when booting nvim it runs git status --porcelain on all projects, which of course gets slower the more projects you have. I've already optimized this to only update the git status of the current project when opening the picker, so once nvim is open there are no performance issues related to maintaining current git status indicators. I think the next optimization is to store the git status of each repo in a file, then only update the file as you enter and exit projects, this would prevent wastefully checking the status of each repo during init, but this starts to become a headache when you consider that the user can modify files and run git commands outside of nvim, though I personally don't. For now I've just left it, as it runs fast enough on my machine. Due to the slowdown when booting nvim, I've defaulted the git_status option to false.

I would also like to add a display for the number of commits ahead of and behind the origin each local repo is, that should probably look exactly the way it does in lazygit. I'll leave that for a future PR, I've already got something functional here.
![image](https://github.com/user-attachments/assets/aeef1ee8-2d98-473a-af91-44fb773e0883)

This is my first PR, any feedback would be greatly appreciated. Let me know if I need to tweak something to get this merged. I don't want to be stuck pointing to my local fork in my nvim config forever.

coffebar commented:
"*" sign is not intuitive for me, you know it can confuse user. With your approach to this feature, nvim users probably want a configurable format like this:
https://github.com/spaceship-prompt/spaceship-prompt/issues/525#issuecomment-1494355353

However, I think this kinda statusline is not in place here and it will consume a lot of visual space and processing time. 

The telescope has a feature to show the preview column when you scroll the list. Maybe we can put a detailed git info to this second column. It has enough space to make clear text messages. Including the active git branch, count of untracked files, etc. If it will be fetched dynamically, on user action, should not be performance issues, I hope. But honestly, I had some projects where git operations could take a while.

What do you think about my idea to show detailed git info in the preview column?

gusjengis commented:
Damn good idea. I still think it's useful to be able to see which projects need attending to at glance, but I agree that an asterisk isn't particularly clear. The preview is a much cleaner place to put this kind of information. 

Are you not worried about giving the telescope picker more features than the others? I'm not totally sure, but I don't think you can do previews with the built-in option, it may be possible with fzf-lua though.

I had to just retype that stuff from memory, I responded hours ago but I must not have forgotten to hit comment. I've since started playing with previews, and I've got telescope showing a view of the contents of each project folder to start. Going to try to get the formatting to look just like oil.nvim, fancy icons and all. Also a good place to put the git status info, though I think they should be separate options, folder previews and git status previews that is. I guess I've now expanded my scope to providing comprehensive project previews.

More to come.

coffebar commented:
I can't select a branch of your PR, please submit a new one with that branch

coffebar commented:
I looked at that branch, it is not ready for merge. It contains stuff related to git status. You need to clean it first.

gusjengis commented:
Good catch, fixed it now. Had to figure out how to use diff view. Compared my branch to your current main, removed everything irrelevant, should be good to go.

https://github.com/coffebar/neovim-project/pull/52

Going to close this for now, I'll open a similar PR later.
